Userland/Libraries/LibPDF/Value.h

```diff
@@ -9,171 +9,74 @@
 #include <AK/Format.h>
 #include <AK/RefPtr.h>
 #include <AK/String.h>
+#include <AK/Variant.h>
 #include <LibPDF/Forward.h>
 #include <LibPDF/Object.h>
 #include <LibPDF/Reference.h>
 
 namespace PDF {
 
-class Value {
+class Value : public Variant<Empty, std::nullptr_t, bool, int, float, Reference, NonnullRefPtr<Object>> {
 public:
-    Value()
-        : m_type(Type::Empty)
-    {
-    }
-
-    struct NullTag {
-    };
-
-    Value(NullTag)
-        : m_type(Type::Null)
-    {
-    }
-
-    Value(bool b)
-        : m_type(Type::Bool)
-    {
-        m_as_bool = b;
-    }
-
-    Value(int i)
-        : m_type(Type::Int)
-    {
-        m_as_int = i;
-    }
-
-    Value(float f)
-        : m_type(Type::Float)
-    {
-        m_as_float = f;
-    }
-
-    Value(Reference ref)
-        : m_type(Type::Ref)
-    {
-        m_as_ref = ref;
-    }
+    using Variant::Variant;
 
-    template<IsObject T>
-    Value(RefPtr<T> obj)
-        : m_type(obj ? Type::Object : Type::Empty)
-    {
-        if (obj) {
-            m_as_object = obj;
-        }
-    }
+    Value()
+        : Variant(Empty()) {};
 
+    // This is a hack: We create a temporary Variant, and call the move-constructor as the super-constructor.
     template<IsObject T>
-    Value(NonnullRefPtr<T> obj)
-        : m_type(Type::Object)
-    {
-        m_as_object = obj;
-    }
-
-    Value(Value const& other)
-    {
-        *this = other;
-    }
-
-    ~Value() = default;
+    Value(RefPtr<T> const& refptr)
+        : Variant(refptr ? Variant(NonnullRefPtr<Object>(*refptr)) : Variant(nullptr)) {};
```

Comment:
I'll repeat the "how about assigning nullptr by default and setting it to *refptr in the ctor body" comment.

Also, extraneous semicolon, and please put the braces on a new line.

Reply:
Gah! I did exactly that, but pushed the wrong branch.

@extraneous semicolon: Indeed! Huh, why didn't clang-format pick that one up?